components/infra/postgres/init.sql

```
@@ -1,19 +1,36 @@
+-- (SQLSTATE 3D000). The sidecar closes that gap; every statement below is
+-- guarded so re-applying against a populated cluster provisions only what is
+-- missing and never errors on what already exists.
 
-SELECT pg_create_physical_replication_slot('replication_slot');
-SELECT * FROM pg_create_logical_replication_slot('logical_slot', 'pgoutput');
+-- Replication role for the physical/logical replica. The password literal here
+-- must match REPLICATION_PASSWORD in .env (the replica authenticates with that
+-- env value) — keep the two in sync; the guard only checks role existence, so
+-- changing one without the other silently breaks replica auth.
+SELECT 'CREATE USER replicator WITH REPLICATION LOGIN ENCRYPTED PASSWORD ''replicator_password'''
+  WHERE NOT EXISTS (SELECT FROM pg_roles WHERE rolname = 'replicator')\gexec
```

SQLFluff (4.2.2)

[error] The 'WHERE' keyword should always start a new line.

(LT14)